### ISSUE TITLE
Modify `DeepLearningModel` class to pass it the loss function, optimizer and metrics by parameters

### ISSUE THREAD

**Issue by gegose** [sherpaai/Sherpa.ai-Federated-Learning-Framework#182]
We could modify `DeepLearningModel` class to pass it the loss function, optimizer and metrics by parameters and then make the `model.compile()` inside the class. The `device` parameter is not adaptable, since to run on GPU, TF and PyTorch work differently and that parameter is specifically for that.

With these changes, we may have to modify the `deepcopy` method to apply the `model.compile()` when copying a model. First of all, we should copy all the attributes and then perform the `model.compile()`.

So we can create a task to adapt `DeepLearningModel` class to the same format. I think that the task would be composed of:

1. Modify `__init__()` and `deepcopy` methods for DeepLearningModel class (I'm not sure if more methods are necessary).
2. Modify all notebooks where `DeepLearningModel`class appears.
3. Modify tests for obtain 100% coverage.

Regards.

_Originally posted by @josebummer in https://github.com/sherpaai/Sherpa.ai-Federated-Learning-Framework/pull/177#discussion_r518183792_
